modules/ballerina-logging/src/main/java/org/ballerinalang/logging/util/BLogLevelMapper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.logging.util;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A mapper class to map log levels from JDK logging to our custom log levels
+ *
+ * @since 0.89
+ */
+public class BLogLevelMapper {
+
+    private static final Map<Level, String> LEVEL_STRING_MAP = Collections.unmodifiableMap(Stream.of(
```

Comment:
Can't we use a switch-case for this? Would be much faster than a HashMap lookup.